sdk/messaging/azservicebus/message.go

```diff
@@ -270,6 +287,15 @@ func newReceivedMessage(amqpMsg *amqp.Message) *ReceivedMessage {
 			msg.TransactionPartitionKey = to.StringPtr(viaPartitionKey.(string))
 		}
 
+		switch asInt64(amqpMsg.Annotations[messageStateAnnotation], 0) {
+		case 1:
+			msg.State = MessageStateDeferred
+		case 2:
+			msg.State = MessageStateScheduled
+		default:
```

Comment:
Is it possible for more states to be added?  If so, they'd fall into this clause until we updated it.

Reply:
This is the "official" list, so I think the fall through will be unexercised (this matches the .net logic, btw).